python-sdk/src/astro/databases/google/bigquery.py

```diff
@@ -59,8 +59,14 @@
 }
 BIGQUERY_WRITE_DISPOSITION = {"replace": "WRITE_TRUNCATE", "append": "WRITE_APPEND"}
 
-NATIVE_AUTODETECT_SCHEMA_SUPPORTED_FILE_TYPES = {FileType.CSV, FileType.NDJSON}
-NATIVE_AUTODETECT_SCHEMA_SUPPORTED_FILE_LOCATIONS = {FileLocation.GS}
+NATIVE_AUTODETECT_SCHEMA_CONFIG: Mapping[
```

Comment:
>Currently, the way we are autodetecting schema for bigquery is loading an entire file, and then truncating the table loaded earlier is not an optimal approach.

Do we not need to fix it for other DBs too?

Reply:
@kaxil  Actually no, this is just a problem for Bigquery. Specifically GCS to Bigquery.

Reply:
Snowflake does not load the data in the table, just create the table in the autodetection step.

cc: @sunank200 @tatiana

Reply:
That's correct this case is only valid for BigQuery. Other databases require the creation of tables explicitly.